### ISSUE TITLE
[ESXi] Creating virtual machine fails (PolicyIDByVirtualMachine)

### ISSUE THREAD

**Issue by w-andre** [hashicorp/terraform-provider-vsphere#1033]
I tried to create a virtual machine on an ESXi server (no vCenter) and received the following error message:
```
Error: this operation is only supported on vCenter: RESOURCE (52), ACTION (PolicyIDByVirtualMachine)

  on test.tf line 46, in resource "vsphere_virtual_machine" "vm":
  46: resource "vsphere_virtual_machine" "vm" {
```
I think the problem is in the following line;
https://github.com/terraform-providers/terraform-provider-vsphere/blob/1c52bb0d1124d759df766edf47e4fe1442fc9a96/vsphere/resource_vsphere_virtual_machine.go#L453

Reading a storage policy id will always fail without vCenter since it is not supported.

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version / vSphere Provider Version

```
Terraform v0.12.24
+ provider.vsphere v1.17.1
```

### Affected Resource(s)
- `vsphere_virtual_machine`

### Terraform Configuration Files

```hcl
...
resource "vsphere_virtual_machine" "vm" {
  name             = "terraform-test"
  resource_pool_id = data.vsphere_resource_pool.pool.id
  datastore_id     = data.vsphere_datastore.datastore.id

  num_cpus                   = 2
  memory                     = 1024
  guest_id                   = "ubuntu64Guest"
  wait_for_guest_net_timeout = 0
  firmware                   = "efi"


  network_interface {
    network_id = data.vsphere_network.network.id
  }

  disk {
    label            = "disk0"
    size             = 20
    thin_provisioned = false
  }
}
```

### Debug Output (extract)

`2020/04/11 10:46:43 [DEBUG] vsphere_virtual_machine.vm: apply errored, but we're indicating that via the Error pointer rather than returning it: this operation is only supported on vCenter: RESOURCE (52), ACTION (PolicyIDByVirtualMachine)`

### Expected Behavior
`terraform apply` performs the action successfully

### Actual Behavior
`terraform apply` fails

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

- using ESXi without vCenter

**Comment by jlundy2** [comment]
Reverting to `v1.15.0` looks to resolve this specific issue.

**Comment by nebula2** [comment]
I can not reproduce this. Did you try turning it off and on again?

**Comment by stuart-c** [comment]
I am also seeing this issue. Reverting to 1.15.0 also stopped the error

**Comment by einyx** [comment]
Error: disk.0: disk label or name must be defined and cannot be computed

  Switching to 1.15 shows the real error for me:

```
on main.tf line 48, in resource "vsphere_virtual_machine" "vm":
  48: resource "vsphere_virtual_machine" "vm" {

```

**Comment by arun12nura** [comment]
> Reverting to `v1.15.0` looks to resolve this specific issue.

@jlundy2  - Can you please me how to revert the plugin to v1.15.0. I am working with windows environment.

**Comment by mjmenger** [comment]
> > Reverting to `v1.15.0` looks to resolve this specific issue.
> 
> @jlundy2 - Can you please me how to revert the plugin to v1.15.0. I am working with windows environment.
```
provider "vsphere" {
  user           = "..."
  password       = "..."
  vsphere_server = "..."
  version = "1.15.0"
}
```
and re-run ```terraform init`` should work. Though you'll likely have issues if you have an existing state file.

**Comment by sumitmaggo** [comment]
Any update on the fix for this issue as this is blocking the use of latest 1.24.1 version for the vsphere provider since the basic operation of launching VM on ESXi is not working?
@bill-rich This commit seems to be causing this issue as storage policy is not supported on ESXi. Can you revert the changes for this commit?
https://github.com/hashicorp/terraform-provider-vsphere/commit/266bcea90f5d5d77f3b1ec66c6bc7cdd54270ba7

**Comment by sumitmaggo** [comment]
@w-andre Just wanted to check if you have been able to get around this issue with the latest 1.24.2 version? Also do we have any update on when this issue will be addressed?

**Comment by parryb** [comment]
Yeah I'm seeing the same error as well which is resolved by using 1.15.0.  Using 1.15.0 does not reveal an underlying error it returns successfully.

**Comment by jmarrnaz** [comment]
So this is still an issue in 1.26.0 . Is there a plan to resolve this for non-vCenter setups? 
Error: this operation is only supported on vCenter: RESOURCE (28), ACTION (PolicyIDByVirtualMachine)

**Comment by comnam90** [comment]
Any progress on this issue?

**Comment by appilon** [comment]
closed via #1408, however as noted there, we don't explicitly support vsphere without vcenter, it is not part of our testing environment, so this fix remains untested.

**Comment by tigerblue77** [comment]
Hi @appilon,
thanks for your update on this. I will test and tell you. Can you just tell me how to use the provider locally by pulling it from this Github ? I tried [this](https://www.reddit.com/r/Terraform/comments/jo0hc9/specifying_a_local_plugin/) but I can't get it to work. Thanks

**Comment by appilon** [comment]
Thanks for offering to test @tigerblue77 . Assuming you are using terraform 0.14+, the easiest way of running the provider locally is to pull the source code and run `go install`. From there you can use a [feature introduced in terraform 0.14](https://www.terraform.io/cli/config/config-file#development-overrides-for-provider-developers) to specify terraform to pickup your local build. You should create a file with the following content:

```
provider_installation {
  dev_overrides {
    "hashicorp/vsphere" = "YourGoPath/bin"
  }

  # For all other providers, install them directly from their origin provider
  # registries as normal. If you omit this, Terraform will _only_ use
  # the dev_overrides block, and so no other providers will be available.
  direct {}
}
```

Then point Terraform to this special config file:
```
$ export TF_CLI_CONFIG_FILE="path/to/config/you/just/made"
```

**Comment by tigerblue77** [comment]
@appilon Thanks for your explanation, it took me a while but I finally got the library working locally and I confirm that the patch works! :D
Do you need any special information as long as I have the test environment on hand and operational? I can take screenshots of whatever you want.
For information, I took back my Terraform VM creation script that was working in v1.15.0 of the plugin, I simply replaced the version with the local version (freshly recovered from the GIT repository) and it works like a charm !

**Comment by appilon** [comment]
Nope, just appreciate someone giving it some confirmation. The scope of this change was very small so a quick check I think is a reasonable.

**Comment by tigerblue77** [comment]
Ok good 👍 have a good evening !

**Comment by jmarrnaz** [comment]
@appilon - I was able to test and thus far looks good in regard to the _PolicyIDByVirtualMachine_ message. As far as the other issue _PolicyIDByVirtual_, I did not experience this issue and I did not see any steps to reproduce. I would be happy to test if you know what it takes to replicate.

It's a great to have this as a resource provider for the ESXI as the ability to spin-up and test in lab environments with TF is literally priceless compared to VMWare vCenter and the licensing that goes with it. Thank you for making these changes. 
I will continue to test as I build out my lab TF deployment and see if anything else comes up. 

Do we expect to have this rolled up into next release?

![Terraform_Plan](https://user-images.githubusercontent.com/57453864/146461203-0c8ea9d6-224d-46e0-9bee-a688a00e3213.jpg)
![Terraform_Apply](https://user-images.githubusercontent.com/57453864/146461242-b933b6ef-3c99-4520-bff2-84fc94164bdf.jpg)
?
